### PR TITLE
WIP: generate default apiserver url for kube2sky

### PR DESCRIPTION
**Pull request by ArtfulCoder**

### CODE REVIEW

**Comment by ArtfulCoder** [comment]
@thockin

**Comment by ArtfulCoder** [comment]
more changes are required.
Please Do Not Merge.

**Comment by ArtfulCoder** [comment]
This PR still does not use https port. Do not merge.

File: cluster/addons/dns/kube2sky/Makefile [hunk 1]
```diff
@@ -4,7 +4,7 @@
 
 .PHONY: all kube2sky container push clean
 
-TAG = 1.4
```
Comment (vishh) [inline cluster/addons/dns/kube2sky/Makefile]: Why update the tag for every PR?

File: cluster/addons/dns/kube2sky/kube2sky.go [hunk 2]
```diff
@@ -48,7 +48,7 @@ var (
 	argEtcdMutationTimeout = flag.Duration("etcd_mutation_timeout", 10*time.Second, "crash after retrying etcd mutation for a specified duration")
 	argEtcdServer          = flag.String("etcd-server", "http://127.0.0.1:4001", "URL to etcd server")
 	argKubecfgFile         = flag.String("kubecfg_file", "", "Location of kubecfg file for access to kubernetes service")
-	argKubeMasterUrl       = flag.String("kube_master_url", "https://${KUBERNETES_SERVICE_HOST}:${KUBERNETES_SERVICE_PORT}", "Url to reach kubernetes master. Env variables in this flag will be expanded.")
+	argKubeMasterUrl       = flag.String("kube_master_url", "", "Url to reach kubernetes master. Env variables in this flag will be expanded.")
```
Comment (vishh) [inline cluster/addons/dns/kube2sky/kube2sky.go]: Why change this flag? If someone wants to use RO service, let them manually override this flag.

File: cluster/addons/dns/kube2sky/kube2sky.go [hunk 3]
```diff
@@ -179,14 +179,43 @@ func getKubeMasterUrl() (string, error) {
 	return parsedUrl.String(), nil
 }
 
+func getDefaultMasterUrl(use_ro_port bool) (string, error) {
```
Comment (vishh) [inline cluster/addons/dns/kube2sky/kube2sky.go]: I find this logic unnecessarily complex. I prefer the original flag with the ability to pass env variables.

**Comment by vishh** [comment]
I don't see why we need this PR. The existing code is functional as-is and we plan to nix RO service support soon.

**Comment by saad-ali** [comment]
Marking WIP as post-1.0

**Comment by ArtfulCoder** [comment]
work has been done in seperate PRs to pass the url from kubeconfig.
closing this PR

File: cluster/addons/dns/kube2sky/kube2sky.go [hunk 4]
```diff
@@ -179,14 +179,43 @@ func getKubeMasterUrl() (string, error) {
...
+	if use_ro_port {
```
Comment (lavalamp) [inline cluster/addons/dns/kube2sky/kube2sky.go]: There's no such thing as the ro port any more, so please don't add this in.

**Pull request closed without merging**